scripts/scil_compute_sf_from_sh.py

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+"""
+Script to sample SF values from a Spherical Harmonics signal. Outputs a Nifti
+file with the SF values and an associated .bvec file with the chosen directions.
+
+If converting from SH to a DWI-like SF volume, --in_bval and --in_b0 need
+to be provided to concatenate the b0 image to the SF, and to generate the new
+bvals file. Otherwise, no .bval file will be created.
+"""
+
+import argparse
+from operator import xor
+
+from dipy.data import SPHERE_FILES, get_sphere
+from dipy.io import read_bvals_bvecs
+from dipy.reconst.shm import order_from_ncoef, sh_to_sf
+import nibabel as nib
+import numpy as np
+
+from scilpy.io.utils import (add_force_b0_arg, add_overwrite_arg,
+                             add_sh_basis_args, assert_inputs_exist,
+                             assert_outputs_exist)
+from scilpy.utils.bvec_bval_tools import (check_b0_threshold)
+
+
+def _build_arg_parser():
+    p = argparse.ArgumentParser(
+        description=__doc__, formatter_class=argparse.RawTextHelpFormatter)
+    p.add_argument('in_sh',
+                   help='Path of the SH volume.')
+    p.add_argument('out_sf',
+                   help='Name of the output SF file to save (bvals/bvecs will '
+                        'be automatically named when necessary).')
+
+    # Optional arguments to generate a DWI-like volume
+    p.add_argument('--in_bval',
+                   help='Optional b-value file, in FSL format, used to '
+                        'generate a DWI-like volume + bvals/bvecs.')
+    p.add_argument('--in_b0',
+                   help='Optional b0 volume, to concatenate to the '
+                        'final SF volume if a DWI representation is needed.')
+
+    # Sphere choice for SF
+    p.add_argument('--sphere', default='repulsion724',
+                   choices=sorted(SPHERE_FILES.keys()),
+                   help='Sphere used for the SH to SF projection. '
+                        '[%(default)s]')
+    add_sh_basis_args(p)
+
+    add_overwrite_arg(p)
+    add_force_b0_arg(p)
+
+    return p
+
+
+def main():
+    parser = _build_arg_parser()
+    args = parser.parse_args()
+
+    assert_inputs_exist(parser, [args.in_sh])
+
+    out_bvecs = args.out_sf.replace(".nii.gz", ".bvec")
+    assert_outputs_exist(parser, args, [args.out_sf, out_bvecs])
+
+    # Either both --in_bval and --in_b0 are provided, or none
+    if xor(bool(args.in_bval), bool(args.in_b0)):
```

Comment:
is there something from argparse that can be use as a group instead of these lines

Reply:
I could put a subparser, which would be used like this : `scil_compute_sf_from_sh.py sh.nii.gz sf.nii.gz dwi_like --bval dwi.bval --b0 b0.nii.gz`